Integration tests fail when check retries

This check command was actually successful, but because it retried, the integration test is considered a failure. Is there something that we can do to prevent the integration tests failing in cases like this?

```
--- FAIL: TestCheckProxy (10.23s)
	install_test.go:237: Received unexpected output
		Expected:
		kubernetes-api: can initialize the client..................................[ok]
		kubernetes-api: can query the Kubernetes API...............................[ok]
		kubernetes-api: is running the minimum Kubernetes API version..............[ok]
		linkerd-api: control plane namespace exists................................[ok]
		linkerd-api: control plane pods are ready..................................[ok]
		linkerd-api: can initialize the client.....................................[ok]
		linkerd-api: can query the control plane API...............................[ok]
		linkerd-api[kubernetes]: control plane can talk to Kubernetes..............[ok]
		linkerd-api[prometheus]: control plane can talk to Prometheus..............[ok]
		linkerd-data-plane: data plane namespace exists............................[ok]
		linkerd-data-plane: data plane proxies are ready...........................[ok]
		linkerd-data-plane: data plane proxy metrics are present in Prometheus.....[ok]
		linkerd-version: can determine the latest version..........................[ok]
		linkerd-version: cli is up-to-date.........................................[ok]
		linkerd-version: data plane is up-to-date..................................[ok]
		
		Status check results are [ok]
		
		Actual:
		kubernetes-api: can initialize the client..................................[ok]
		kubernetes-api: can query the Kubernetes API...............................[ok]
		kubernetes-api: is running the minimum Kubernetes API version..............[ok]
		linkerd-api: control plane namespace exists................................[ok]
		linkerd-api: control plane pods are ready..................................[ok]
		linkerd-api: can initialize the client.....................................[ok]
		linkerd-api: can query the control plane API...............................[ok]
		linkerd-api[kubernetes]: control plane can talk to Kubernetes..............[ok]
		linkerd-api[prometheus]: control plane can talk to Prometheus..............[ok]
		linkerd-data-plane: data plane namespace exists............................[ok]
		linkerd-data-plane: data plane proxies are ready...........................[retry] -- The "linkerd-proxy" container in the "ver-stacked-tls-smoke-test/smoke-test-gateway-796c7c56d-lzr7j" pod is not ready
		linkerd-data-plane: data plane proxies are ready...........................[retry] -- The "linkerd-proxy" container in the "ver-stacked-tls-smoke-test/smoke-test-gateway-796c7c56d-lzr7j" pod is not ready
		linkerd-data-plane: data plane proxies are ready...........................[ok]
		linkerd-data-plane: data plane proxy metrics are present in Prometheus.....[ok]
		linkerd-version: can determine the latest version..........................[ok]
		linkerd-version: cli is up-to-date.........................................[ok]
		linkerd-version: data plane is up-to-date..................................[ok]
		
		Status check results are [ok]
```

--- a/test/install_test.go
+++ b/test/install_test.go
@@ -131,7 +131,12 @@ func TestCheckPostInstall(t *testing.T) {
 	var out string
 	var err error
 	overallErr := TestHelper.RetryFor(30*time.Second, func() error {
-		out, _, err = TestHelper.LinkerdRun("check", "--expected-version", TestHelper.GetVersion())
+		out, _, err = TestHelper.LinkerdRun(
+			"check",
+			"--expected-version",
+			TestHelper.GetVersion(),
+			"--wait=0",
+			)
 		return err
 	})
 	if overallErr != nil {
@@ -220,14 +225,21 @@ func TestInject(t *testing.T) {
 
 func TestCheckProxy(t *testing.T) {
 	prefixedNs := TestHelper.GetTestNamespace("smoke-test")
-	out, _, err := TestHelper.LinkerdRun(
-		"check",
-		"--proxy",
-		"--expected-version",
-		TestHelper.GetVersion(),
-		"--namespace",
-		prefixedNs,
-	)
+	var out string
+	err := TestHelper.RetryFor(2*time.Minute, func() error {
+		var err error
+		out, _, err = TestHelper.LinkerdRun(
+			"check",
+			"--proxy",
+			"--expected-version",
+			TestHelper.GetVersion(),
+			"--namespace",
+			prefixedNs,
+			"--wait=0",
+		)
+		return err
+	})
+
 	if err != nil {
 		t.Fatalf("Check command failed\n%s", out)
 	}

